src/Apps/Artwork/Components/ArtworkDetails/__tests__/ArtworkDetails.test.tsx

```diff
@@ -184,5 +208,84 @@ describe("ArtworkDetails", () => {
       })
       expect(wrapper).toBeTruthy()
     })
+
+    it("shows the to 'Log in to request report' button if feature enabled, artwork is biddable, and unauthenticated", async () => {
+      mockEnableRequestConditionReport.mockReturnValue(true)
+
+      wrapper = await getWrapper({
+        ...ArtworkDetailsFixture,
+        isBiddable: true,
+        saleArtwork: {
+          internalID: "sale-artwork-id",
+          id: "sale-artwork-id",
+        },
+      })
+      expect(wrapper.html()).toMatch("to request")
```

Comment:
Any reason to not assert upon the full text "Log in to request report"? Would there be a way to add a data attribute to an element to make it easier to target a subset of text on the page?

Reply:
The design comps have "Log in" in the button and "to request" next to the button. In the rendered html, you won't see the full text of "Log in to request" as one chunk. I tried matching upon the presence of certain components like `UnauthenticatedContent` to complement the string matching but was having some trouble there.

Not sure I follow the data attribute idea. Can you elaborate?

Reply:
I believe he means something like
```
div.wrapper data-type='condition-report-request'
  button
    Log in
  p
    to request
```

Reply:
> Not sure I follow the data attribute idea. Can you elaborate?

Sure – there might not be a comparable pattern with React, but when testing Rails apps, there's often a pattern of doing something like

```html
<div data-test="some-selector">
  <p>foo</p>
  <p>bar</p>
</div> 
<p>baz</p>
```

so that one can select the text within some selector like the following:

(pseudo-ruby / capybara)
```ruby
within '.some-selector' do
  text = page.text()
end
```

---

There might something similar here could be employed!

Reply:
Ahh, gotcha. Yes, I'm familiar with "within" for capybara in ruby test suites. I'll see what's available here to better scope the assertion.

Reply:
This spec is skipped because because, as a QueryRenderer, this component's render is deferred to the client. I think this would be better tested as a full relay and interaction-enabled spec backed by our `createTestEnv` tooling.